### PR TITLE
customizable component colors

### PR DESCRIPTION
**Pull request by R3dst0ne**
Add functionality requested in #1232.

---
There are some topics that need discussion:
- [ ] radix color (currently blue)
- [ ] ghost color (currently gray)
- [x] Icons
- [ ] [ComponentDrawContext.java](https://github.com/R3dst0ne/logisim-evolution/blob/component-colors/src/main/java/com/cburch/logisim/comp/ComponentDrawContext.java)
- [ ] secondary color used (currently gray)
   - [ ] arithmetic inputs e.g. carry
   - [ ] select circle for plexers 
   - [ ] R and S input for D-/T-/J-/RS-FF
- [ ] dark gray of DIP, LED Bar, LED Matrix, port I/O (, TTL?)

### CODE REVIEW

**Review by maehne — COMMENTED** [review]
Looks so far good to me. Thanks @R3dst0ne!

**Comment by R3dst0ne** [comment]
It is not perfect but IMO it sets a good baseline.

File: src/main/java/com/cburch/logisim/std/ttl/AbstractTtlGate.java
Comment (MarcinOrlowski) [inline src/main/java/com/cburch/logisim/std/ttl/AbstractTtlGate.java:225]: Either one-liner or multi-liner but then with mandatory brackets, please.

File: src/main/java/com/cburch/logisim/std/wiring/Pin.java
Comment (MarcinOrlowski) [inline src/main/java/com/cburch/logisim/std/wiring/Pin.java:1052]: `var` or `final var` please.

File: src/main/java/com/cburch/logisim/std/wiring/PullResistor.java
Comment (MarcinOrlowski) [inline src/main/java/com/cburch/logisim/std/wiring/PullResistor.java:114]: `var` or `final var` please

File: src/main/java/com/cburch/logisim/tools/AddTool.java
Comment (MarcinOrlowski) [inline src/main/java/com/cburch/logisim/tools/AddTool.java:193]: Can you please correct variable name to not start with capital letter?

File: src/main/resources/resources/logisim/strings/gui/gui_pl.properties
Comment (MarcinOrlowski) [inline src/main/resources/resources/logisim/strings/gui/gui_pl.properties:616]:
```
windowComponentColor = Kolor komponentu
```

Comment (MarcinOrlowski) [reply src/main/resources/resources/logisim/strings/gui/gui_pl.properties:616]: With `:` if needed etc.

**Review by MarcinOrlowski — CHANGES_REQUESTED** [review]
Also can we have these `// WIP` comments either replaced with more descriptive ones  (i.e. `// WIP, Ticket #XYZ @R3dst0ne` or removed?

File: src/main/java/com/cburch/logisim/prefs/AppPreferences.java
Comment (R3dst0ne) [reply src/main/java/com/cburch/logisim/prefs/AppPreferences.java:514]: They were the same color until now, and nobody complained.
So I don't see a problem with this.

**Comment by R3dst0ne** [comment]
Should be ready for merging.

**Comment by MarcinOrlowski** [comment]
Also all checks are back alive so that's not a stopper anymore.

File: src/main/java/com/cburch/logisim/prefs/AppPreferences.java
Comment (MarcinOrlowski) [reply src/main/java/com/cburch/logisim/prefs/AppPreferences.java:514]: Fair enough.

File: src/main/java/com/cburch/logisim/std/ttl/AbstractTtlGate.java
Comment (MarcinOrlowski) [reply src/main/java/com/cburch/logisim/std/ttl/AbstractTtlGate.java:225]: And this is marked resolved yet I do not see any change?

**Comment by R3dst0ne** [comment]
@MarcinOrlowski I addressed all your change requests. Could you please do, what ever is necessary, so that this PR is not blocked anymore?

**Review by maehne — CHANGES_REQUESTED** [review]
Thanks for your work @R3dst0ne and sorry for the delay reviewing your PR. The proposed changes look sensible to me. I have some questions:

- [ ] Could you please check tha the recently merged PRs with  additional TTL components are also covered by your PR?
- [ ] Did you check how this PR behaves together with a dark theme?

File: src/main/java/com/cburch/logisim/gui/icons/CounterIcon.java
Comment (maehne) [reply src/main/java/com/cburch/logisim/gui/icons/CounterIcon.java:29]: @R3dst0ne: I think this should be checked how it looks with a dark theme and fixed before merging if there's some readability issue.

File: src/main/java/com/cburch/logisim/prefs/AppPreferences.java
Comment (R3dst0ne) [reply src/main/java/com/cburch/logisim/prefs/AppPreferences.java:514]: Marcin did ask the same question.
See my response to [his comment](https://github.com/logisim-evolution/logisim-evolution/pull/1487#discussion_r1012687462).

File: gradle.properties
Comment (R3dst0ne) [reply gradle.properties]: reversed

File: src/main/java/com/cburch/logisim/prefs/AppPreferences.java
Comment (maehne) [reply src/main/java/com/cburch/logisim/prefs/AppPreferences.java:514]: I am fine with your response/decision to leave it as is.

File: src/main/java/com/cburch/logisim/gui/icons/CounterIcon.java
Comment (R3dst0ne) [reply src/main/java/com/cburch/logisim/gui/icons/CounterIcon.java]: Now it works with all themes.
Additionally, I removed some dead functionality introduced in #784.

**Comment by R3dst0ne** [comment]
> I have some questions:
> 
> - [ ]  Could you please check tha the recently merged PRs with  additional TTL components are also covered by your PR?
> - [ ]  Did you check how this PR behaves together with a dark theme?

I checked all the recently added TTL components.

I did test it with different themes, and it does improve the readability (in dark ones) quite a lot.

File: src/main/java/com/cburch/logisim/gui/main/Selection.java
Comment (R3dst0ne) [reply src/main/java/com/cburch/logisim/gui/main/Selection.java:70]: Honestly, I don't remember.
I'm not even sure what this code is for.
If you could tell me what this code is for, maybe it will help my memory. ^^'

Comment (maehne) [reply src/main/java/com/cburch/logisim/gui/main/Selection.java:70]: If you don't remember then it is probably best to simply remove the WIP comments that don't provide any details.

**Review by maehne — DISMISSED** [review]
Good for me except for that one WIP comment without any details. Just remove it if you think it is of no importance anymore.

**Comment by R3dst0ne** [comment]
The Code style linter failed in the `Lint changed files` stage.
It produced the following error massage:
```
Run # Run CheckStyle
Downloading checkstyle-10.3.4-all.jar...
Error: Process completed with exit code 8.
```
Is there something I can do do fix this?

**Comment by davidhutchens** [comment]
I had a similarly odd failure in #1646. It later worked. My best guess as to that problem is that the github server failed to deliver our source. This problem seems to be that, again, the github server has failed to deliver something. I'm not sure what is going on, but I doubt you can fix it. At least, it does not seem to be related to what you did in this PR. It is possible that making an irrelevant change to force it to check again may bring success.

**Review by maehne — APPROVED** [review]
Looks good to me?

@MarcinOrlowski: What is left from your side?

It would be nice to get this PR finally merged.

File: src/main/java/com/cburch/logisim/std/ttl/AbstractTtlGate.java
Comment (R3dst0ne) [reply src/main/java/com/cburch/logisim/std/ttl/AbstractTtlGate.java:225]: This got resolved a while ago, it seems that GitHub does not want to recognize this.
Take a look at the latest to commits.

**Comment by maehne** [comment]
@MarcinOrlowski: What is still blocking this PR from being merged?

**Comment by R3dst0ne** [comment]
@MarcinOrlowski
Sorry for mentioning you again, but we have been waiting for an answer for two months now.
Could you please approve this PR or leave a comment on what in your opinion is still missing?

**Review by maehne — APPROVED** [review]
@BFH-ktt1: I propose to merge this PR. If @MarcinOrlowski would like to improve it further, he can open a PR to address these issues once he finds time.

**Pull request merged**
